src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -998,9 +1016,19 @@ private void ApplyDoUpdate(DoUpdate update)
                         Definition: updateDefn,
                         Args: argsForUpdate,
                         Headers: update.Headers));
+                    LinkedListNode<(string Name, string Id)>? warnableExecution = null;
+                    if (updateDefn.UnfinishedPolicy == HandlerUnfinishedPolicy.WarnAndAbandon)
+                    {
+                        warnableExecution = warnableInProgressUpdates.AddLast(
+                            (update.Name, update.Id));
+                    }
```

Comment:
IMO I would _always_ track these, and have this if be around the actual warn rather than adding to the list.

This also better supports a future `GetUnfinishedHandlers` API

Reply:
IMO no need to retain extra unused information for .NET IMO (they are memory conscious, I almost waited until was first accessed to create the linked list, but figured it was trivial). This internal stuff is easy to refactor when that future occurs.

Reply:
Ok, I have realized the `AllHandlersFinished` call was not meant to be warnable only.

Reply:
Also just realistically this is a vanishingly small amount of memory

Reply:
Yeah, it was more about lac of necessity. But it is there now.